diff -urp a/opt/atlassian/jira/bin/setenv.sh b/opt/atlassian/jira/bin/setenv.sh
--- a/opt/atlassian/jira/bin/setenv.sh	2018-04-11 18:12:12.953249348 +0800
+++ b/opt/atlassian/jira/bin/setenv.sh	2018-04-11 18:12:26.825374848 +0800
@@ -1,18 +1,18 @@
 #
 # One way to set the JIRA HOME path is here via this variable.  Simply uncomment it and set a valid path like /jira/home.  You can of course set it outside in the command terminal.  That will also work.
 #
-#JIRA_HOME=""
+JIRA_HOME="${JIRA_HOME:-/var/atlassian/application-data/jira}"
 
 #
 #  Occasionally Atlassian Support may recommend that you set some specific JVM arguments.  You can use this variable below to do that.
 #
-JVM_SUPPORT_RECOMMENDED_ARGS=""
+JVM_SUPPORT_RECOMMENDED_ARGS="${JVM_SUPPORT_RECOMMENDED_ARGS:-"-Datlassian.plugins.enable.wait=300"}"
 
 #
 # The following 2 settings control the minimum and maximum given to the JIRA Java virtual machine.  In larger JIRA instances, the maximum amount will need to be increased.
 #
-JVM_MINIMUM_MEMORY="384m"
-JVM_MAXIMUM_MEMORY="768m"
+JVM_MINIMUM_MEMORY="${JVM_MINIMUM_MEMORY:-384m}"
+JVM_MAXIMUM_MEMORY="${JVM_MAXIMUM_MEMORY:-768m}"
 
 #
 # The following are the required arguments for JIRA.
@@ -34,7 +34,7 @@ JVM_REQUIRED_ARGS='-Djava.awt.headless=t
 # Prevents the JVM from suppressing stack traces if a given type of exception
 # occurs frequently, which could make it harder for support to diagnose a problem.
 #-----------------------------------------------------------------------------------
-JVM_EXTRA_ARGS="-XX:-OmitStackTraceInFastThrow"
+JVM_EXTRA_ARGS="-XX:-OmitStackTraceInFastThrow -Dcatalina.connector.scheme=${CATALINA_CONNECTOR_SCHEME:-http} -Dcatalina.connector.secure=${CATALINA_CONNECTOR_SECURE:-false} -Dcatalina.connector.proxyName=${CATALINA_CONNECTOR_PROXYNAME:-} -Dcatalina.connector.proxyPort=${CATALINA_CONNECTOR_PROXYPORT:-} -Dcatalina.context.path=${CATALINA_CONTEXT_PATH:-}"
 
 PRGDIR=`dirname "$0"`
 cat "${PRGDIR}"/jirabanner.txt
@@ -43,12 +43,12 @@ JIRA_HOME_MINUSD=""
 if [ "$JIRA_HOME" != "" ]; then
     echo $JIRA_HOME | grep -q " "
     if [ $? -eq 0 ]; then
-	    echo ""
-	    echo "--------------------------------------------------------------------------------------------------------------------"
-		echo "   WARNING : You cannot have a JIRA_HOME environment variable set with spaces in it.  This variable is being ignored"
-	    echo "--------------------------------------------------------------------------------------------------------------------"
+        echo ""
+        echo "--------------------------------------------------------------------------------------------------------------------"
+        echo "   WARNING : You cannot have a JIRA_HOME environment variable set with spaces in it.  This variable is being ignored"
+        echo "--------------------------------------------------------------------------------------------------------------------"
     else
-		JIRA_HOME_MINUSD=-Djira.home=$JIRA_HOME
+        JIRA_HOME_MINUSD=-Djira.home=$JIRA_HOME
     fi
 fi
 
@@ -77,16 +77,16 @@ fi
 export CATALINA_PID
 
 if [ -z "$CATALINA_BASE" ]; then
-  if [ -z "$CATALINA_HOME" ]; then
-    LOGBASE=$PRGDIR
-    LOGTAIL=..
-  else
-    LOGBASE=$CATALINA_HOME
-    LOGTAIL=.
-  fi
+    if [ -z "$CATALINA_HOME" ]; then
+        LOGBASE=$PRGDIR
+        LOGTAIL=..
+    else
+        LOGBASE=$CATALINA_HOME
+        LOGTAIL=.
+    fi
 else
-  LOGBASE=$CATALINA_BASE
-  LOGTAIL=.
+    LOGBASE=$CATALINA_BASE
+    LOGTAIL=.
 fi
 
 PUSHED_DIR=`pwd`
diff -urp a/opt/atlassian/jira/conf/server.xml b/opt/atlassian/jira/conf/server.xml
--- a/opt/atlassian/jira/conf/server.xml	2018-04-11 18:12:01.589146745 +0800
+++ b/opt/atlassian/jira/conf/server.xml	2018-04-11 18:14:49.512927988 +0800
@@ -33,6 +33,7 @@
 
         <Connector port="8080" maxThreads="150" minSpareThreads="25" connectionTimeout="20000" enableLookups="false"
                    maxHttpHeaderSize="8192" protocol="HTTP/1.1" useBodyEncodingForURI="true" redirectPort="8443"
+                   scheme="${catalina.connector.scheme}" secure="${catalina.connector.secure}" proxyName="${catalina.connector.proxyName}" proxyPort="${catalina.connector.proxyPort}"
                    acceptCount="100" disableUploadTimeout="true" bindOnInit="false"/>
 
         <!--
@@ -95,7 +96,7 @@
         <Engine name="Catalina" defaultHost="localhost">
             <Host name="localhost" appBase="webapps" unpackWARs="true" autoDeploy="true">
 
-                <Context path="" docBase="${catalina.home}/atlassian-jira" reloadable="false" useHttpOnly="true">
+                <Context path="${catalina.context.path}" docBase="${catalina.home}/atlassian-jira" reloadable="false" useHttpOnly="true">
                     <Resource name="UserTransaction" auth="Container" type="javax.transaction.UserTransaction"
                               factory="org.objectweb.jotm.UserTransactionFactory" jotm.timeout="60"/>
                     <Manager pathname=""/>
